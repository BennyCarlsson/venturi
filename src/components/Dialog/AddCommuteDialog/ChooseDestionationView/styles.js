import styled from "styled-components";
import Button from "components/Button";
import Heading from "components/Heading";
import Paragraph from "components/Paragraph";
import ClosedIconButton from "components/CloseIconButton";

export const TopCTAButtonTitleSubtitle = styled.div`
  margin-top: 8px;
  padding-bottom: 48px;
`;
export const CTAButtonTitleWrapper = styled.div`
  display: flex;
  align-items: baseline;
  margin-bottom: 16px;
`;
export const StyledToParagraph = styled(Paragraph)`
  padding-top: 14px;
`;

export const StyledTitleHeading = styled(Heading)`
  margin-left: 24px;
`;

export const InputDropDownWrapper = styled.div`
  position: relative;
`;

export const StyledButton = styled(Button)`
  margin-bottom: 56px;
`;

export const StyledClosedIconButton = styled(ClosedIconButton)``;
