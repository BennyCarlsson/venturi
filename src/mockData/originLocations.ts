export const locations = {
  locations: {
    loadingOrigin: false,
    loadingDestination: false,
    destinationLocations: [
      {
        id: '9021014014010000',
        idx: '1',
        name: 'Hålta kyrka, Kungälv',
        lat: '57.895206',
        lon: '11.827647',
        type: 0,
      },
      {
        id: '9021014012743000',
        idx: '2',
        name: 'Gårda, Mölndal',
        lat: '57.565347',
        lon: '12.092784',
        type: 0,
      },
      {
        id: '9021014002515000',
        idx: '3',
        name: 'Toftaåsgatan, Göteborg',
        lat: '57.645953',
        lon: '11.917170',
        type: 0,
      },
      {
        id: '9021014015934000',
        idx: '4',
        name: 'Hagaån, Orust',
        lat: '58.139155',
        lon: '11.625866',
        type: 0,
      },
      {
        id: '9021014021670000',
        idx: '5',
        name: 'Råane bro, Uddevalla',
        lat: '58.293814',
        lon: '11.984122',
        type: 0,
      },
      {
        id: '9021014021673000',
        idx: '6',
        name: 'Råane skola, Uddevalla',
        lat: '58.288996',
        lon: '11.988985',
        type: 0,
      },
      {
        id: '9021014021672000',
        idx: '7',
        name: 'Råane södra, Uddevalla',
        lat: '58.291360',
        lon: '11.986549',
        type: 0,
      },
      {
        id: 'A U Bååths väg, 439 94 Onsala',
        idx: '8',
        name: 'A U Bååths väg, 439 94 Onsala',
        lat: '57.390569',
        lon: '12.022587',
        type: 1,
      },
      {
        id: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        idx: '9',
        name: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        lat: '59.908834',
        lon: '12.510476',
        type: 1,
      },
      {
        id: 'Almvägen, 432 79 Tvååker',
        idx: '10',
        name: 'Almvägen, 432 79 Tvååker',
        lat: '57.044575',
        lon: '12.408116',
        type: 1,
      },
      {
        id: 'Alvägen, 432 79 Tvååker',
        idx: '11',
        name: 'Alvägen, 432 79 Tvååker',
        lat: '57.042903',
        lon: '12.404539',
        type: 1,
      },
      {
        id: 'Amandas väg, 432 77 Tvååker',
        idx: '12',
        name: 'Amandas väg, 432 77 Tvååker',
        lat: '57.060683',
        lon: '12.529336',
        type: 1,
      },
      {
        id: 'Anders Kristians väg, 432 77 Tvååker',
        idx: '13',
        name: 'Anders Kristians väg, 432 77 Tvååker',
        lat: '57.073115',
        lon: '12.368690',
        type: 1,
      },
      {
        id: 'Anemonvägen, 432 76 Tvååker',
        idx: '14',
        name: 'Anemonvägen, 432 76 Tvååker',
        lat: '57.012168',
        lon: '12.352572',
        type: 1,
      },
      {
        id: 'Angryd, 432 77 Tvååker',
        idx: '15',
        name: 'Angryd, 432 77 Tvååker',
        lat: '57.089359',
        lon: '12.540393',
        type: 1,
      },
      {
        id: 'Annelundsvägen, 432 76 Tvååker',
        idx: '16',
        name: 'Annelundsvägen, 432 76 Tvååker',
        lat: '57.042570',
        lon: '12.315941',
        type: 1,
      },
      {
        id: 'Askvägen, 432 79 Tvååker',
        idx: '17',
        name: 'Askvägen, 432 79 Tvååker',
        lat: '57.043370',
        lon: '12.403046',
        type: 1,
      },
      {
        id: 'Aspvägen, 432 79 Tvååker',
        idx: '18',
        name: 'Aspvägen, 432 79 Tvååker',
        lat: '57.044215',
        lon: '12.406894',
        type: 1,
      },
      {
        id: 'Augusts väg, 432 77 Tvååker',
        idx: '19',
        name: 'Augusts väg, 432 77 Tvååker',
        lat: '57.058598',
        lon: '12.526792',
        type: 1,
      },
      {
        id: 'Avavägen, 432 78 Tvååker',
        idx: '20',
        name: 'Avavägen, 432 78 Tvååker',
        lat: '57.047721',
        lon: '12.397797',
        type: 1,
      },
      {
        id: 'Rör, 432 77 Tvååker',
        idx: '21',
        name: 'Rör, 432 77 Tvååker',
        lat: '57.059937',
        lon: '12.445323',
        type: 1,
      },
      {
        id: 'Sik, 432 76 Tvååker',
        idx: '22',
        name: 'Sik, 432 76 Tvååker',
        lat: '57.001220',
        lon: '12.367845',
        type: 1,
      },
      {
        id: 'Påarps gård, Håcksvik',
        idx: '23',
        name: 'Påarps gård, Håcksvik',
        lat: '57.315473',
        lon: '13.147488',
        type: 1,
      },
      {
        id: 'Ängarna, 432 76 Tvååker',
        idx: '24',
        name: 'Ängarna, 432 76 Tvååker',
        lat: '57.028376',
        lon: '12.375692',
        type: 1,
      },
      {
        id: 'Ängvägen, 432 78 Tvååker',
        idx: '25',
        name: 'Ängvägen, 432 78 Tvååker',
        lat: '57.047217',
        lon: '12.402363',
        type: 1,
      },
      {
        id: 'Ästad, 432 77 Tvååker',
        idx: '26',
        name: 'Ästad, 432 77 Tvååker',
        lat: '57.066535',
        lon: '12.562155',
        type: 1,
      },
      {
        id: 'Ästad bondgård, 432 77 Tvååker',
        idx: '27',
        name: 'Ästad bondgård, 432 77 Tvååker',
        lat: '57.063245',
        lon: '12.555953',
        type: 1,
      },
      {
        id: 'Bäckvägen, 432 79 Tvååker',
        idx: '28',
        name: 'Bäckvägen, 432 79 Tvååker',
        lat: '57.038741',
        lon: '12.404889',
        type: 1,
      },
      {
        id: 'Ballstorp Kullaås, 535 93 Kvänum',
        idx: '29',
        name: 'Ballstorp Kullaås, 535 93 Kvänum',
        lat: '58.191256',
        lon: '13.253812',
        type: 1,
      },
      {
        id: 'Barrvägen, 432 79 Tvååker',
        idx: '30',
        name: 'Barrvägen, 432 79 Tvååker',
        lat: '57.035675',
        lon: '12.405806',
        type: 1,
      },
      {
        id: 'Beckasinvägen, 432 76 Tvååker',
        idx: '31',
        name: 'Beckasinvägen, 432 76 Tvååker',
        lat: '57.007296',
        lon: '12.358487',
        type: 1,
      },
      {
        id: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        idx: '32',
        name: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        lat: '59.421358',
        lon: '14.248622',
        type: 1,
      },
      {
        id: 'Bjällemad, 432 79 Tvååker',
        idx: '33',
        name: 'Bjällemad, 432 79 Tvååker',
        lat: '57.037329',
        lon: '12.411271',
        type: 1,
      },
      {
        id: 'Bjällemadvägen, 432 79 Tvååker',
        idx: '34',
        name: 'Bjällemadvägen, 432 79 Tvååker',
        lat: '57.037662',
        lon: '12.406678',
        type: 1,
      },
      {
        id: 'Bjären, 432 76 Tvååker',
        idx: '35',
        name: 'Bjären, 432 76 Tvååker',
        lat: '57.056485',
        lon: '12.351331',
        type: 1,
      },
      {
        id: 'Björka, 432 77 Tvååker',
        idx: '36',
        name: 'Björka, 432 77 Tvååker',
        lat: '57.086932',
        lon: '12.592782',
        type: 1,
      },
      {
        id: 'Björkäng, 432 76 Tvååker',
        idx: '37',
        name: 'Björkäng, 432 76 Tvååker',
        lat: '57.008672',
        lon: '12.362631',
        type: 1,
      },
      {
        id: 'Björkängs camping, 432 76 Tvååker',
        idx: '38',
        name: 'Björkängs camping, 432 76 Tvååker',
        lat: '57.009606',
        lon: '12.350226',
        type: 1,
      },
      {
        id: 'Björkängs havsbad, 432 76 Tvååker',
        idx: '39',
        name: 'Björkängs havsbad, 432 76 Tvååker',
        lat: '57.009418',
        lon: '12.350792',
        type: 1,
      },
      {
        id: 'Björkängs kvarn, 432 76 Tvååker',
        idx: '40',
        name: 'Björkängs kvarn, 432 76 Tvååker',
        lat: '57.013822',
        lon: '12.357813',
        type: 1,
      },
      {
        id: 'Björkängsvägen, 432 78 Tvååker',
        idx: '41',
        name: 'Björkängsvägen, 432 78 Tvååker',
        lat: '57.039127',
        lon: '12.398336',
        type: 1,
      },
      {
        id: 'Björkedalsvägen, 432 77 Tvååker',
        idx: '42',
        name: 'Björkedalsvägen, 432 77 Tvååker',
        lat: '57.072153',
        lon: '12.369822',
        type: 1,
      },
      {
        id: 'Björktrastvägen, 432 76 Tvååker',
        idx: '43',
        name: 'Björktrastvägen, 432 76 Tvååker',
        lat: '57.010676',
        lon: '12.356177',
        type: 1,
      },
      {
        id: 'Bjurbäcken Braåsen, 688 91 Storfors',
        idx: '44',
        name: 'Bjurbäcken Braåsen, 688 91 Storfors',
        lat: '59.577204',
        lon: '14.211325',
        type: 1,
      },
      {
        id: 'Blidsbergs Påarp, 523 77 Blidsberg',
        idx: '45',
        name: 'Blidsbergs Påarp, 523 77 Blidsberg',
        lat: '57.941051',
        lon: '13.480934',
        type: 1,
      },
      {
        id: 'Blomvägen, 432 79 Tvååker',
        idx: '46',
        name: 'Blomvägen, 432 79 Tvååker',
        lat: '57.043586',
        lon: '12.401392',
        type: 1,
      },
      {
        id: 'Bocklerud Laåsen, 672 94 Årjäng',
        idx: '47',
        name: 'Bocklerud Laåsen, 672 94 Årjäng',
        lat: '59.393186',
        lon: '11.994550',
        type: 1,
      },
      {
        id: 'Bofinksvägen, 432 76 Tvååker',
        idx: '48',
        name: 'Bofinksvägen, 432 76 Tvååker',
        lat: '57.009660',
        lon: '12.353300',
        type: 1,
      },
      {
        id: 'Borrås, 432 76 Tvååker',
        idx: '49',
        name: 'Borrås, 432 76 Tvååker',
        lat: '57.014434',
        lon: '12.441340',
        type: 1,
      },
      {
        id: 'Bosgårdsvägen, 432 79 Tvååker',
        idx: '50',
        name: 'Bosgårdsvägen, 432 79 Tvååker',
        lat: '57.044970',
        lon: '12.399747',
        type: 1,
      },
    ],
    originLocations: [
      {
        id: '9021014014010000',
        idx: '1',
        name: 'Hålta kyrka, Kungälv',
        lat: '57.895206',
        lon: '11.827647',
        type: 0,
      },
      {
        id: '9021014012743000',
        idx: '2',
        name: 'Gårda, Mölndal',
        lat: '57.565347',
        lon: '12.092784',
        type: 0,
      },
      {
        id: '9021014002515000',
        idx: '3',
        name: 'Toftaåsgatan, Göteborg',
        lat: '57.645953',
        lon: '11.917170',
        type: 0,
      },
      {
        id: '9021014015934000',
        idx: '4',
        name: 'Hagaån, Orust',
        lat: '58.139155',
        lon: '11.625866',
        type: 0,
      },
      {
        id: '9021014021670000',
        idx: '5',
        name: 'Råane bro, Uddevalla',
        lat: '58.293814',
        lon: '11.984122',
        type: 0,
      },
      {
        id: '9021014021673000',
        idx: '6',
        name: 'Råane skola, Uddevalla',
        lat: '58.288996',
        lon: '11.988985',
        type: 0,
      },
      {
        id: '9021014021672000',
        idx: '7',
        name: 'Råane södra, Uddevalla',
        lat: '58.291360',
        lon: '11.986549',
        type: 0,
      },
      {
        id: 'A U Bååths väg, 439 94 Onsala',
        idx: '8',
        name: 'A U Bååths väg, 439 94 Onsala',
        lat: '57.390569',
        lon: '12.022587',
        type: 1,
      },
      {
        id: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        idx: '9',
        name: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        lat: '59.908834',
        lon: '12.510476',
        type: 1,
      },
      {
        id: 'Almvägen, 432 79 Tvååker',
        idx: '10',
        name: 'Almvägen, 432 79 Tvååker',
        lat: '57.044575',
        lon: '12.408116',
        type: 1,
      },
      {
        id: 'Alvägen, 432 79 Tvååker',
        idx: '11',
        name: 'Alvägen, 432 79 Tvååker',
        lat: '57.042903',
        lon: '12.404539',
        type: 1,
      },
      {
        id: 'Amandas väg, 432 77 Tvååker',
        idx: '12',
        name: 'Amandas väg, 432 77 Tvååker',
        lat: '57.060683',
        lon: '12.529336',
        type: 1,
      },
      {
        id: 'Anders Kristians väg, 432 77 Tvååker',
        idx: '13',
        name: 'Anders Kristians väg, 432 77 Tvååker',
        lat: '57.073115',
        lon: '12.368690',
        type: 1,
      },
      {
        id: 'Anemonvägen, 432 76 Tvååker',
        idx: '14',
        name: 'Anemonvägen, 432 76 Tvååker',
        lat: '57.012168',
        lon: '12.352572',
        type: 1,
      },
      {
        id: 'Angryd, 432 77 Tvååker',
        idx: '15',
        name: 'Angryd, 432 77 Tvååker',
        lat: '57.089359',
        lon: '12.540393',
        type: 1,
      },
      {
        id: 'Annelundsvägen, 432 76 Tvååker',
        idx: '16',
        name: 'Annelundsvägen, 432 76 Tvååker',
        lat: '57.042570',
        lon: '12.315941',
        type: 1,
      },
      {
        id: 'Askvägen, 432 79 Tvååker',
        idx: '17',
        name: 'Askvägen, 432 79 Tvååker',
        lat: '57.043370',
        lon: '12.403046',
        type: 1,
      },
      {
        id: 'Aspvägen, 432 79 Tvååker',
        idx: '18',
        name: 'Aspvägen, 432 79 Tvååker',
        lat: '57.044215',
        lon: '12.406894',
        type: 1,
      },
      {
        id: 'Augusts väg, 432 77 Tvååker',
        idx: '19',
        name: 'Augusts väg, 432 77 Tvååker',
        lat: '57.058598',
        lon: '12.526792',
        type: 1,
      },
      {
        id: 'Avavägen, 432 78 Tvååker',
        idx: '20',
        name: 'Avavägen, 432 78 Tvååker',
        lat: '57.047721',
        lon: '12.397797',
        type: 1,
      },
      {
        id: 'Rör, 432 77 Tvååker',
        idx: '21',
        name: 'Rör, 432 77 Tvååker',
        lat: '57.059937',
        lon: '12.445323',
        type: 1,
      },
      {
        id: 'Sik, 432 76 Tvååker',
        idx: '22',
        name: 'Sik, 432 76 Tvååker',
        lat: '57.001220',
        lon: '12.367845',
        type: 1,
      },
      {
        id: 'Påarps gård, Håcksvik',
        idx: '23',
        name: 'Påarps gård, Håcksvik',
        lat: '57.315473',
        lon: '13.147488',
        type: 1,
      },
      {
        id: 'Ängarna, 432 76 Tvååker',
        idx: '24',
        name: 'Ängarna, 432 76 Tvååker',
        lat: '57.028376',
        lon: '12.375692',
        type: 1,
      },
      {
        id: 'Ängvägen, 432 78 Tvååker',
        idx: '25',
        name: 'Ängvägen, 432 78 Tvååker',
        lat: '57.047217',
        lon: '12.402363',
        type: 1,
      },
      {
        id: 'Ästad, 432 77 Tvååker',
        idx: '26',
        name: 'Ästad, 432 77 Tvååker',
        lat: '57.066535',
        lon: '12.562155',
        type: 1,
      },
      {
        id: 'Ästad bondgård, 432 77 Tvååker',
        idx: '27',
        name: 'Ästad bondgård, 432 77 Tvååker',
        lat: '57.063245',
        lon: '12.555953',
        type: 1,
      },
      {
        id: 'Bäckvägen, 432 79 Tvååker',
        idx: '28',
        name: 'Bäckvägen, 432 79 Tvååker',
        lat: '57.038741',
        lon: '12.404889',
        type: 1,
      },
      {
        id: 'Ballstorp Kullaås, 535 93 Kvänum',
        idx: '29',
        name: 'Ballstorp Kullaås, 535 93 Kvänum',
        lat: '58.191256',
        lon: '13.253812',
        type: 1,
      },
      {
        id: 'Barrvägen, 432 79 Tvååker',
        idx: '30',
        name: 'Barrvägen, 432 79 Tvååker',
        lat: '57.035675',
        lon: '12.405806',
        type: 1,
      },
      {
        id: 'Beckasinvägen, 432 76 Tvååker',
        idx: '31',
        name: 'Beckasinvägen, 432 76 Tvååker',
        lat: '57.007296',
        lon: '12.358487',
        type: 1,
      },
      {
        id: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        idx: '32',
        name: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        lat: '59.421358',
        lon: '14.248622',
        type: 1,
      },
      {
        id: 'Bjällemad, 432 79 Tvååker',
        idx: '33',
        name: 'Bjällemad, 432 79 Tvååker',
        lat: '57.037329',
        lon: '12.411271',
        type: 1,
      },
      {
        id: 'Bjällemadvägen, 432 79 Tvååker',
        idx: '34',
        name: 'Bjällemadvägen, 432 79 Tvååker',
        lat: '57.037662',
        lon: '12.406678',
        type: 1,
      },
      {
        id: 'Bjären, 432 76 Tvååker',
        idx: '35',
        name: 'Bjären, 432 76 Tvååker',
        lat: '57.056485',
        lon: '12.351331',
        type: 1,
      },
      {
        id: 'Björka, 432 77 Tvååker',
        idx: '36',
        name: 'Björka, 432 77 Tvååker',
        lat: '57.086932',
        lon: '12.592782',
        type: 1,
      },
      {
        id: 'Björkäng, 432 76 Tvååker',
        idx: '37',
        name: 'Björkäng, 432 76 Tvååker',
        lat: '57.008672',
        lon: '12.362631',
        type: 1,
      },
      {
        id: 'Björkängs camping, 432 76 Tvååker',
        idx: '38',
        name: 'Björkängs camping, 432 76 Tvååker',
        lat: '57.009606',
        lon: '12.350226',
        type: 1,
      },
      {
        id: 'Björkängs havsbad, 432 76 Tvååker',
        idx: '39',
        name: 'Björkängs havsbad, 432 76 Tvååker',
        lat: '57.009418',
        lon: '12.350792',
        type: 1,
      },
      {
        id: 'Björkängs kvarn, 432 76 Tvååker',
        idx: '40',
        name: 'Björkängs kvarn, 432 76 Tvååker',
        lat: '57.013822',
        lon: '12.357813',
        type: 1,
      },
      {
        id: 'Björkängsvägen, 432 78 Tvååker',
        idx: '41',
        name: 'Björkängsvägen, 432 78 Tvååker',
        lat: '57.039127',
        lon: '12.398336',
        type: 1,
      },
      {
        id: 'Björkedalsvägen, 432 77 Tvååker',
        idx: '42',
        name: 'Björkedalsvägen, 432 77 Tvååker',
        lat: '57.072153',
        lon: '12.369822',
        type: 1,
      },
      {
        id: 'Björktrastvägen, 432 76 Tvååker',
        idx: '43',
        name: 'Björktrastvägen, 432 76 Tvååker',
        lat: '57.010676',
        lon: '12.356177',
        type: 1,
      },
      {
        id: 'Bjurbäcken Braåsen, 688 91 Storfors',
        idx: '44',
        name: 'Bjurbäcken Braåsen, 688 91 Storfors',
        lat: '59.577204',
        lon: '14.211325',
        type: 1,
      },
      {
        id: 'Blidsbergs Påarp, 523 77 Blidsberg',
        idx: '45',
        name: 'Blidsbergs Påarp, 523 77 Blidsberg',
        lat: '57.941051',
        lon: '13.480934',
        type: 1,
      },
      {
        id: 'Blomvägen, 432 79 Tvååker',
        idx: '46',
        name: 'Blomvägen, 432 79 Tvååker',
        lat: '57.043586',
        lon: '12.401392',
        type: 1,
      },
      {
        id: 'Bocklerud Laåsen, 672 94 Årjäng',
        idx: '47',
        name: 'Bocklerud Laåsen, 672 94 Årjäng',
        lat: '59.393186',
        lon: '11.994550',
        type: 1,
      },
      {
        id: 'Bofinksvägen, 432 76 Tvååker',
        idx: '48',
        name: 'Bofinksvägen, 432 76 Tvååker',
        lat: '57.009660',
        lon: '12.353300',
        type: 1,
      },
      {
        id: 'Borrås, 432 76 Tvååker',
        idx: '49',
        name: 'Borrås, 432 76 Tvååker',
        lat: '57.014434',
        lon: '12.441340',
        type: 1,
      },
      {
        id: 'Bosgårdsvägen, 432 79 Tvååker',
        idx: '50',
        name: 'Bosgårdsvägen, 432 79 Tvååker',
        lat: '57.044970',
        lon: '12.399747',
        type: 1,
      },
    ],
  },
}
