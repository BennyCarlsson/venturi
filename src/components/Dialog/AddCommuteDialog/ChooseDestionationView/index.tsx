import Paragraph from 'components/Paragraph';
import { useAppDispatch, useAppSelector } from 'hooks/redux';
import React, { useCallback, useRef, useState } from 'react';
import {
  fetchDestinationLocationsOnName,
  fetchOriginLocationsOnName,
  Location
} from 'redux/locationsSlice';
import { setCurrentTable } from 'redux/tripSlice';
import DropDown from '../DropDown';
import SearchInput from '../SearchInput';
import { saveOriginAndDestination } from 'utils/localStorage';
import { StyledButton, StyledParagraph, StyledTitleHeading, StyledToParagraph } from './styles';

let timeout: any;
const debounce = (fn: Function, delay: number) => {
  if (timeout !== undefined) {
    clearTimeout(timeout);
  }
  timeout = setTimeout(fn, delay);
};

type ChooseDestionationViewProps = {};

const ChooseDestionationView = ({ ...props }: ChooseDestionationViewProps) => {
  const [originInput, setOriginInput] = useState('');
  const [showOriginDropDown, setShowOriginDropDown] = useState(false);
  const [destinationInput, setDestinationInput] = useState('');
  const [showDestinationDropDown, setShowDestinationDropDown] = useState(false);
  const [originLocation, setOriginLocation] = useState<Location | null>(null);
  const [destinationLocation, setDestinationLocation] = useState<Location | null>(null);

  const dispatch = useAppDispatch();
  const originLocations = useAppSelector((state) => state.locations.originLocations);
  const destinationLocations = useAppSelector((state) => state.locations.destinationLocations);
  const loading = useAppSelector((state) => state.locations.loadingDestination);
  const error = useAppSelector((state) => state.locations.destinationError);
  let promise = useRef<any>();

  const originInputOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value;
    setOriginInput(value);
    debounce(() => {
      requestOrigin(value);
    }, 500);
  };

  const requestOrigin = useCallback(
    (value: string) => {
      if (value.length > 1) {
        if (promise.current && !error && loading) {
          promise.current.abort();
        }
        setShowOriginDropDown(true);
        promise.current = dispatch(fetchOriginLocationsOnName(value));
      } else {
        setShowOriginDropDown(false);
      }
    },
    [dispatch, error, loading]
  );

  const destinationOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value;
    setDestinationInput(value);
    debounce(() => {
      requestDestination(value);
    }, 500);
  };
  const requestDestination = useCallback(
    (value: string) => {
      if (value.length > 1) {
        if (promise.current && !error && loading) {
          promise.current.abort();
        }
        setShowDestinationDropDown(true);
        promise.current = dispatch(fetchDestinationLocationsOnName(value));
      } else {
        setShowDestinationDropDown(false);
      }
    },
    [dispatch, error, loading]
  );
  return (
    <React.Fragment {...props}>
      <span>
        <StyledTitleHeading headingType={'h2'} fontSize={28}>
          Destination
        </StyledTitleHeading>
        <StyledParagraph fontSize={14}>
          Välj destinationer för din resa. Du kan alltid ändra eller lägga till flera vid ett senare
          tillfälle.
        </StyledParagraph>
        <Paragraph fontSize={18}>Från</Paragraph>
        <SearchInput
          value={originInput}
          onChange={originInputOnChange}
          setShowDropDown={setShowOriginDropDown}
        />
        <DropDown
          locations={originLocations}
          setInput={setOriginInput}
          setLocation={setOriginLocation}
          show={showOriginDropDown}
        />
        <StyledToParagraph fontSize={18}>Till</StyledToParagraph>
        <SearchInput
          value={destinationInput}
          onChange={destinationOnChange}
          setShowDropDown={setShowDestinationDropDown}
        />
        <DropDown
          locations={destinationLocations}
          setInput={setDestinationInput}
          setLocation={setDestinationLocation}
          show={showDestinationDropDown}
        />
      </span>
      <StyledButton
        onClick={() => {
          if (originLocation && destinationLocation)
            dispatch(setCurrentTable({ origin: originLocation, destination: destinationLocation }));
          saveOriginAndDestination({
            origin: originLocation as Location,
            destination: destinationLocation as Location
          });
        }}
      >
        Namnge din resa
      </StyledButton>
      {error && <h2>{error}</h2>}
    </React.Fragment>
  );
};

export default ChooseDestionationView;
