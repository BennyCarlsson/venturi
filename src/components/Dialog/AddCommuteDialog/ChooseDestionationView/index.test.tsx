import ChooseDestionationView from '.';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { useDispatch } from 'react-redux';
import { locations } from 'mockData/originLocations';
import { render, screen, waitFor, waitForElementToBeRemoved } from 'testUtils/test-utils';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';

jest.mock('react-redux', () => ({
  ...(jest.requireActual('react-redux') as {}),
  useDispatch: jest.fn()
}));

describe('<ChooseDestionationView />', () => {
  const useAppDispatch = jest.fn();
  beforeEach(() => {
    (useDispatch as jest.Mock).mockReturnValue(useAppDispatch);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<ChooseDestionationView />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    render(<ChooseDestionationView />, {});
    expect(screen).toBeTruthy();
  });

  it('should show dropdown when typing in the from input', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('from-input'), 'aa');
    await waitFor(() => expect(screen.getByRole('list')).toBeInTheDocument());
    expect(screen.getAllByRole('listitem').length).toBe(locations.locations.originLocations.length);
  });

  it('should show locations when typing in the from input', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('from-input'), 'aa');
    await waitFor(() => expect(screen.getByRole('list')).toBeInTheDocument());
    const testLocation1Name = locations.locations.originLocations[0].name;
    const testLocation2Name =
      locations.locations.originLocations[locations.locations.originLocations.length - 1].name;
    expect(screen.getByText(testLocation1Name)).toBeInTheDocument();
    expect(screen.getByText(testLocation2Name)).toBeInTheDocument();
  });

  it('should hide dropdown when only 1 char in input field (from input)', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('from-input'), 'aa');
    await waitFor(() => expect(screen.getByRole('list')).toBeInTheDocument());
    expect(screen.getAllByRole('listitem').length).toBe(locations.locations.originLocations.length);
    userEvent.type(screen.getByTestId('from-input'), '{backspace}');
    await waitForElementToBeRemoved(screen.getByRole('list'));
  });

  it('should not show dropdown when only typed in one char (from input)', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('from-input'), 'a');
    await act(async () => {
      await new Promise((r) => setTimeout(r, 600));
    });

    expect(screen.queryByRole('list')).not.toBeInTheDocument();
  });

  it('should set input to location onClick and call dispatch (from input)', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('from-input'), 'aa');
    await waitFor(() => expect(screen.getByRole('list')).toBeInTheDocument());
    userEvent.click(screen.getAllByRole('listitem')[0]);
    await act(async () => {
      await new Promise((r) => setTimeout(r, 100));
    });
    const locationName = locations.locations.originLocations[0].name;
    expect(screen.getByTestId('from-input')).toHaveValue(locationName);
    expect(screen.queryByRole('list')).not.toBeInTheDocument();
  });

  it('should show dropdown when typing in the to input', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('to-input'), 'aa');
    await waitFor(() => expect(screen.getByRole('list')).toBeInTheDocument());
    expect(screen.getAllByRole('listitem').length).toBe(
      locations.locations.destinationLocations.length
    );
  });

  it('should show locations when typing in the to input', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('to-input'), 'aa');
    await waitFor(() => expect(screen.getByRole('list')).toBeInTheDocument());
    const testLocation1Name = locations.locations.destinationLocations[0].name;
    const testLocation2Name =
      locations.locations.destinationLocations[locations.locations.destinationLocations.length - 1]
        .name;
    expect(screen.getByText(testLocation1Name)).toBeInTheDocument();
    expect(screen.getByText(testLocation2Name)).toBeInTheDocument();
  });

  it('should hide dropdown when only 1 char in input field (to input)', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('to-input'), 'aa');
    await waitFor(() => expect(screen.getByRole('list')).toBeInTheDocument());
    expect(screen.getAllByRole('listitem').length).toBe(
      locations.locations.destinationLocations.length
    );
    userEvent.type(screen.getByTestId('to-input'), '{backspace}');
    await waitForElementToBeRemoved(screen.getByRole('list'));
  });

  it('should not show dropdown when only typed in one char (to input)', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('to-input'), 'a');
    await act(async () => {
      await new Promise((r) => setTimeout(r, 600));
    });

    expect(screen.queryByRole('list')).not.toBeInTheDocument();
  });

  it('should set input to location onClick and call dispatch (to input)', async () => {
    render(<ChooseDestionationView />, { initialState: { ...locations } });
    userEvent.type(screen.getByTestId('to-input'), 'aa');
    await waitFor(() => expect(screen.getByRole('list')).toBeInTheDocument());
    userEvent.click(screen.getAllByRole('listitem')[0]);
    await act(async () => {
      await new Promise((r) => setTimeout(r, 100));
    });
    const locationName = locations.locations.destinationLocations[0].name;
    expect(screen.getByTestId('to-input')).toHaveValue(locationName);
    expect(screen.queryByRole('list')).not.toBeInTheDocument();
  });
});
