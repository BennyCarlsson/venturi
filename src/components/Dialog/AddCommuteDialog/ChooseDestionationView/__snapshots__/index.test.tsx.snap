// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseDestionationView /> should match snapshot 1`] = `
<DocumentFragment>
  <span>
    <h2
      class="sc-gKsewC icwmdU sc-iqHYGH bVotdH"
      font-size="28"
    >
      Destination
    </h2>
    <p
      class="sc-bdfBwQ QuMsF sc-crrsfI dipcGx"
      font-size="14"
    >
      Välj destinationer för din resa. Du kan alltid ändra eller lägga till flera vid ett senare tillfälle.
    </p>
    <p
      class="sc-bdfBwQ kMVXxV"
      font-size="18"
    >
      Från
    </p>
    <input
      class="sc-hKgILt jroXQp"
      data-testid="from-input"
      value=""
    />
    <p
      class="sc-bdfBwQ kMVXxV sc-kEjbxe hVozjA"
      font-size="18"
    >
      Till
    </p>
    <input
      class="sc-hKgILt jroXQp"
      data-testid="to-input"
      value=""
    />
  </span>
  <button
    class="sc-eCssSg bljGiV sc-dQppl etkeeH"
  >
    Namnge din resa
  </button>
</DocumentFragment>
`;
